internal/service/bcmdataexports/export.go

```diff
@@ -0,0 +1,473 @@
+			err.Error(),
+		)
+		return
+	}
+}
+
+func (r *resourceExport) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+const (
+	statusHealthy = "Healthy"
+	statusError   = "Unhealthy"
+)
+
+func waitExportCreated(ctx context.Context, conn *bcmdataexports.Client, id string, timeout time.Duration) (*bcmdataexports.GetExportOutput, error) {
+	stateConf := &retry.StateChangeConf{
+		Pending:                   []string{},
+		Target:                    []string{statusHealthy},
+		Refresh:                   statusExport(ctx, conn, id),
+		Timeout:                   timeout,
+		NotFoundChecks:            20,
+		ContinuousTargetOccurence: 2,
+	}
+
+	outputRaw, err := stateConf.WaitForStateContext(ctx)
+	if out, ok := outputRaw.(*bcmdataexports.GetExportOutput); ok {
+		return out, err
+	}
+
+	return nil, err
+}
+
+func waitExportUpdated(ctx context.Context, conn *bcmdataexports.Client, id string, timeout time.Duration) (*bcmdataexports.GetExportOutput, error) {
+	stateConf := &retry.StateChangeConf{
+		Pending:                   []string{statusError},
+		Target:                    []string{statusHealthy},
+		Refresh:                   statusExport(ctx, conn, id),
+		Timeout:                   timeout,
+		NotFoundChecks:            20,
+		ContinuousTargetOccurence: 2,
+	}
+
+	outputRaw, err := stateConf.WaitForStateContext(ctx)
+	if out, ok := outputRaw.(*bcmdataexports.GetExportOutput); ok {
+		return out, err
+	}
+
+	return nil, err
+}
+
+func statusExport(ctx context.Context, conn *bcmdataexports.Client, id string) retry.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		out, err := findExportByID(ctx, conn, id)
+		if tfresource.NotFound(err) {
+			return nil, "", nil
+		}
+
+		if err != nil {
+			return nil, "", err
+		}
+
+		return out, statusHealthy, nil
+	}
+}
+
+func findExportByID(ctx context.Context, conn *bcmdataexports.Client, exportArn string) (*bcmdataexports.GetExportOutput, error) {
+	in := &bcmdataexports.GetExportInput{
+		ExportArn: aws.String(exportArn),
+	}
+
+	out, err := conn.GetExport(ctx, in)
+	if errs.IsA[*awstypes.ResourceNotFoundException](err) {
+		return nil, &retry.NotFoundError{
+			LastError:   err,
+			LastRequest: in,
+		}
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	if out == nil {
+		return nil, tfresource.NewEmptyResultError(in)
+	}
+
+	return out, nil
+}
+
+type resourceExportData struct {
+	Export   fwtypes.ListNestedObjectValueOf[ExportData] `tfsdk:"export"`
+	ID       types.String                                `tfsdk:"id"`
+	Tags     types.Map                                   `tfsdk:"tags"`
+	TagsAll  types.Map                                   `tfsdk:"tags_all"`
+	Timeouts timeouts.Value                              `tfsdk:"timeouts"`
+}
+
+type ExportData struct {
```

Comment:
We may not need to export this struct if it is not used outside of this package. Similar for other exported structs